events/apigw.go

```diff
@@ -133,63 +133,65 @@ type APIGatewayV2HTTPResponse struct {
 }
 
 // APIGatewayWebsocketProxyRequestContext contains the information to identify
 // the AWS account and resources invoking the Lambda function. It also includes
 // Cognito identity information for the caller.
 type APIGatewayWebsocketProxyRequestContext struct {
-	AccountID          string                    `json:"accountId"`
-	ResourceID         string                    `json:"resourceId"`
-	Stage              string                    `json:"stage"`
-	RequestID          string                    `json:"requestId"`
-	Identity           APIGatewayRequestIdentity `json:"identity"`
-	ResourcePath       string                    `json:"resourcePath"`
-	Authorizer         interface{}               `json:"authorizer"`
-	HTTPMethod         string                    `json:"httpMethod"`
-	APIID              string                    `json:"apiId"` // The API Gateway rest API Id
-	ConnectedAt        int64                     `json:"connectedAt"`
-	ConnectionID       string                    `json:"connectionId"`
-	DomainName         string                    `json:"domainName"`
-	Error              string                    `json:"error"`
-	EventType          string                    `json:"eventType"`
-	ExtendedRequestID  string                    `json:"extendedRequestId"`
-	IntegrationLatency string                    `json:"integrationLatency"`
-	MessageDirection   string                    `json:"messageDirection"`
-	MessageID          interface{}               `json:"messageId"`
-	RequestTime        string                    `json:"requestTime"`
-	RequestTimeEpoch   int64                     `json:"requestTimeEpoch"`
-	RouteKey           string                    `json:"routeKey"`
-	Status             string                    `json:"status"`
+	AccountID            string                    `json:"accountId,omitempty"`
+	ResourceID           string                    `json:"resourceId,omitempty"`
+	Stage                string                    `json:"stage"`
+	RequestID            string                    `json:"requestId"`
+	Identity             APIGatewayRequestIdentity `json:"identity"`
+	ResourcePath         string                    `json:"resourcePath,omitempty"`
+	Authorizer           interface{}               `json:"authorizer,omitempty"`
+	HTTPMethod           string                    `json:"httpMethod,omitempty"`
+	APIID                string                    `json:"apiId"` // The API Gateway rest API Id
+	ConnectedAt          int64                     `json:"connectedAt"`
+	ConnectionID         string                    `json:"connectionId"`
+	DomainName           string                    `json:"domainName"`
+	Error                string                    `json:"error,omitempty"`
+	EventType            string                    `json:"eventType"`
+	ExtendedRequestID    string                    `json:"extendedRequestId"`
+	IntegrationLatency   string                    `json:"integrationLatency,omitempty"`
```

Comment:
undo the json tag changes, they're not related the the feature you're trying to add.

Reply:
Thanks for your comment, but it's related to the feature because `integrationLatency` field is not in the WebSocket message sent on the $disconnect route as [@tinhda wrote ](https://github.com/aws/aws-lambda-go/pull/548#discussion_r1465099695).
If I remove `.omitempty` json tag, TestApiGatewayWebsocketRequestDisconnectMarshaling would fail.